src/utils.rs

```diff
@@ -310,3 +309,22 @@ pub(crate) fn process_ready_message(
 
     Ok((messages, is_ready))
 }
+
+////////////////////////////
+// Test Utility Functions //
+////////////////////////////
+#[cfg(test)]
+use rand::{
+    rngs::{OsRng, StdRng},
+    SeedableRng,
+};
+/// Returns an rng to be used for testing. This will print the rng seed
+/// to stderr so that if a test fails, the failing seed can be recovered
+/// and used for debugging.
+#[cfg(test)]
+pub(crate) fn get_test_rng() -> StdRng {
```

Comment:
naming nit: I think [Rust convention names getters without the `get` prefix](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter), e.g. so `get_test_rng()` would be called `test_rng()`. But I'm not sure if that applies to independent functions (as opposed to methods on an object like in the example at the link).

Reply:
I think getters usually imply getting fields from an object. I'm fine with whatever convention we prefer.

Reply:
Cool. Then I leave it to @tyurek to decide.